server/src/main/java/org/opensearch/rest/NamedRoute.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest;
+
+import org.opensearch.OpenSearchException;
+
+/**
+ * A named Route
+ *
+ * @opensearch.api
+ */
+public class NamedRoute extends RestHandler.Route {
+    private static final String VALID_ACTION_NAME_PATTERN = "^[a-zA-Z0-9:/*_]*$";
+    static final int MAX_LENGTH_OF_ACTION_NAME = 250;
+
+    private final String name;
+
+    public boolean isValidRouteName(String routeName) {
+        if (routeName == null || routeName.isBlank()) {
+            return false;
+        }
+        if (routeName.length() > MAX_LENGTH_OF_ACTION_NAME) {
+            return false;
+        }
```

Comment:
Can we merge this condition with the one above?

Reply:
Done